Gemfile

```diff
@@ -1,5 +1,21 @@
-source "https://rubygems.org"
+source 'https://rubygems.org'
 
-gem 'uswds-jekyll', :git => 'https://github.com/18F/uswds-jekyll.git'
-gem 'jekyll_pages_api_search'
-gem 'jekyll-sitemap'
+ruby '2.7.1'
+
+gem 'jekyll'
```

Comment:
Any reason we don't optimistically lock these gems?

Reply:
For the jekyll dependency, the isildur-bane template just relies on [uswds-jekyll to specify what it needs ](https://github.com/18F/uswds-jekyll/blob/0a63e49e32c516c8f31861f25950b60529bf94b4/uswds-jekyll.gemspec#L15).

The path that led me to remove the pessimistic operator for the ruby version is a little more convoluted:
* There's at least 3 environments that a site will get built in: docker using image from hub.docker.com, CircleCI using its own cached/pre-built ruby image, Federalist using rvm. A 4th environment is local ruby, but a guide maintainer is on their own for that.
* not specifying a explicit patch version (e.g., 2.7 instead of 2.7.1) lead to bundler errors over time as different docker images/environments would use different versions of bundler
* for guides that use this template, the goal is to be completely hands off for guide maintainers as far as all these technical bits are concerned. So by specifying a specific ruby version (to the patch number), we have the best chance that if a build was working one week, it will continue to work in weeks to come every time.

The ruby version will get updated in downstream guides through this general process:
1. dependabot notices a new version is available
1. new ruby version is set and tested in docker, CircleCI, Federalist in template repo by one of the Engineering Guild co-leads (i.e., the template repo maintainers)
1. new ruby version is merged in upstream template
1. GitHub action in downstream guide automatically applies the changes made in upstream template repo

At the moment, I've not actually turned on the schedule for the GitHub action – I wanted to see this all work well in a 2nd guide first. So for today, [that last step can be done manually using the Action tab of a guide](https://github.com/18F/development-guide/actions).

References: https://github.com/18F/isildurs-bane/pull/12, https://github.com/18F/isildurs-bane/pull/14